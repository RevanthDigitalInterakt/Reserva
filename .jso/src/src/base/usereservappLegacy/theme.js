  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.theme = undefined;
  var tokens = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[0]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  var theme = exports.theme = {
    space: {
      quarck: 4,
      nano: 8,
      micro: 12,
      xxxs: 16,
      xxs: 24,
      xs: 32,
      sm: 40,
      md: 48,
      xl: 64,
      xxl: 80
    },
    shadows: {
      bottomBarShadow: '0px -3px 2px rgba(0,0,0,0.1)',
      topBarShadow: '0px 3px 2px rgba(0,0,0,0.1)'
    },
    colors: {
      preto: tokens.preto,
      vermelhoFechadoRSV: tokens.vermelhoFechadoRSV,
      marinho: tokens.marinho,
      neutroFrio2: tokens.neutroFrio2,
      neutroQuente2: tokens.neutroQuente2,
      vermelhoRSV: tokens.vermelhoRSV,
      verde: tokens.verde,
      neutroQuente1: tokens.neutroQuente1,
      offWhite: tokens.offWhite,
      offWhite15: tokens.offWhite15,
      whiteSecondary: tokens.whiteSecondary,
      neutroFrio1: tokens.neutroFrio1,
      leve: tokens.leve,
      vermelhoAlerta: tokens.vermelhoAlerta,
      verdeSucesso: tokens.verdeSucesso,
      amareloAtencao: tokens.amareloAtencao,
      dropDownBorderColor: tokens.dropDownBorderColor,
      white: tokens.white,
      fullBlack: tokens.fullBlack,
      divider: tokens.divider,
      backgoundInput: tokens.backgoundInput,
      backgroundApp: tokens.backgroundApp,
      transparente: 'transparent',
      modalBackDropColor: tokens.modalBackDropColor,
      borderButton: tokens.borderButton,
      dividerCounter: tokens.dividerCounter,
      backgoundDivider: tokens.backgoundDivider,
      progressTextColor: tokens.progressTextColor,
      backgroundMenuOpened: tokens.backgroundMenuOpened,
      searchBarTextColor: tokens.searchBarTextColor,
      borderColorProduct: tokens.borderColorProduct
    },
    fonts: {
      nunitoBlack: 'Nunito-Black',
      nunitoBlackItalic: 'Nunito-BlackItalic',
      nunitoBold: 'Nunito-Bold',
      nunitoBoldItalic: 'Nunito-BoldItalic',
      nunitoExtraBold: 'Nunito-ExtraBold',
      nunitoExtraBoldItalic: 'Nunito-ExtraBoldItalic',
      nunitoExtraLight: 'Nunito-ExtraLight',
      nunitoExtraLightItalic: 'Nunito-ExtraLightItalic',
      nunitoItalic: 'Nunito-Italic',
      nunitoLight: 'Nunito-Light',
      nunitoLightItalic: 'Nunito-LightItalic',
      nunitoRegular: 'Nunito-Regular',
      nunitoSemiBold: 'Nunito-SemiBold',
      nunitoSemiBoldItalic: 'Nunito-SemiBoldItalic',
      reservaDisplayRegular: 'ReservaDisplay-Regular',
      reservaSerifBlack: 'ReservaSerif-Black',
      reservaSerifBlackItalic: 'ReservaSerif-BlackItalic',
      reservaSerifBold: 'ReservaSerif-Bold',
      reservaSerifBoldItalic: 'ReservaSerif-BoldItalic',
      reservaSerifItalic: 'ReservaSerif-Italic',
      reservaSerifLight: 'ReservaSerif-Light',
      reservaSerifLightItalic: 'ReservaSerif-LightItalic',
      reservaSerifMedium: 'ReservaSerif-Medium',
      reservaSerifMediumItalic: 'ReservaSerif-MediumItalic',
      reservaSerifRegular: 'ReservaSerif-Regular',
      reservaSerifThin: 'ReservaSerif-Thin',
      reservaSerifThinItalic: 'ReservaSerif-ThinItalic',
      reservaSansBlack: 'ReservaSans-Black',
      reservaSansBlackItalic: 'ReservaSans-BlackItalic',
      reservaSansBold: 'ReservaSans-Bold',
      reservaSansBoldItalic: 'ReservaSans-BoldItalic',
      reservaSansItalic: 'ReservaSans-Italic',
      reservaSansLight: 'ReservaSans-Light',
      reservaSansLightItalic: 'ReservaSans-LightItalic',
      reservaSansMedium: 'ReservaSans-Medium',
      reservaSansMediumItalic: 'ReservaSans-MediumItalic',
      reservaSansRegular: 'ReservaSans-Regular',
      reservaSansThin: 'ReservaSans-Thin',
      reservaSansThinItalic: 'ReservaSans-ThinItalic'
    },
    fontWeights: {
      light: 300,
      normal: 400,
      medium: 500,
      SemiBold: 600,
      bold: 700,
      black: 900
    },
    fontSizes: [4, 8, 12, 14, 16, 18, 20, 24, 27, 32, 48],
    radii: {
      pico: 2,
      nano: 5,
      xxxs: 10,
      xs: 15,
      sm: 20,
      pill: 500,
      infinity: 999999
    },
    borderWidths: {
      none: tokens.borderWidthNone,
      hairline: tokens.borderWidthHairline,
      thin: tokens.borderWidthThin,
      thick: tokens.borderWidthThick,
      heavy: tokens.borderWidthHeavy,
      borderDiscountTag: tokens.borderDiscountTag,
      stepper: 3
    }
  };
