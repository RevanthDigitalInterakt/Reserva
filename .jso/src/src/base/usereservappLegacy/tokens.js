  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.whiteSecondary = exports.white = exports.vermelhoRSV = exports.vermelhoFechadoRSV = exports.vermelhoAlerta = exports.verdeSucesso = exports.verde = exports.searchBarTextColor = exports.progressTextColor = exports.preto = exports.offWhite15 = exports.offWhite = exports.neutroQuente2 = exports.neutroQuente1 = exports.neutroFrio2 = exports.neutroFrio1 = exports.modalBackDropColor = exports.marinho = exports.leve = exports.fullBlack = exports.dropDownBorderColor = exports.dividerCounter = exports.divider = exports.borderWidthThin = exports.borderWidthThick = exports.borderWidthNone = exports.borderWidthHeavy = exports.borderWidthHairline = exports.borderDiscountTag = exports.borderColorProduct = exports.borderButton = exports.backgroundMenuOpened = exports.backgroundApp = exports.backgoundInput = exports.backgoundDivider = exports.amareloAtencao = undefined;
  var white = exports.white = '#FFF';
  var fullBlack = exports.fullBlack = '#000';

  // cores primarias
  var preto = exports.preto = '#333333';
  var vermelhoFechadoRSV = exports.vermelhoFechadoRSV = '#5A0000';
  var marinho = exports.marinho = '#13294B';
  var neutroFrio2 = exports.neutroFrio2 = '#8A8C8E';
  var neutroQuente2 = exports.neutroQuente2 = '#B1ACA8';
  var divider = exports.divider = '#D8D9DA';
  var dividerCounter = exports.dividerCounter = '#EAEAEA';
  var progressTextColor = exports.progressTextColor = '#979797';
  var searchBarTextColor = exports.searchBarTextColor = '#666666';

  // cores secundarias
  var vermelhoRSV = exports.vermelhoRSV = '#A00000';
  var verde = exports.verde = '#22372B';
  var neutroQuente1 = exports.neutroQuente1 = '#D7CFCA';
  var offWhite = exports.offWhite = '#D6D1C4';
  var offWhite15 = exports.offWhite15 = '#F9F8F6';
  var whiteSecondary = exports.whiteSecondary = '#FAFAFA';
  var neutroFrio1 = exports.neutroFrio1 = '#D8D9DA';
  var leve = exports.leve = '#F9E1DD';

  // cores do sistema
  var vermelhoAlerta = exports.vermelhoAlerta = '#EF1E1E';
  var verdeSucesso = exports.verdeSucesso = '#38A238';
  var amareloAtencao = exports.amareloAtencao = '#FEA528';
  var dropDownBorderColor = exports.dropDownBorderColor = '#DDDDDD';

  // larguras de borda
  var borderWidthNone = exports.borderWidthNone = 0;
  var borderWidthHairline = exports.borderWidthHairline = 1;
  var borderWidthThin = exports.borderWidthThin = 2;
  var borderWidthThick = exports.borderWidthThick = 4;
  var borderWidthHeavy = exports.borderWidthHeavy = 8;
  var borderDiscountTag = exports.borderDiscountTag = 12;

  // cores de borda
  var borderButton = exports.borderButton = '#333333';
  var borderColorProduct = exports.borderColorProduct = '#757575';

  // backgrounds
  var backgroundMenuOpened = exports.backgroundMenuOpened = '#F1F1F1';
  var backgroundApp = exports.backgroundApp = '#F9F9F9';
  var modalBackDropColor = exports.modalBackDropColor = '#AFAFAF';
  var backgoundInput = exports.backgoundInput = '#F0F0F0';
  var backgoundDivider = exports.backgoundDivider = '#F5F5F5';
