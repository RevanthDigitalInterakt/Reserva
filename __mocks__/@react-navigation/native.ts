const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
