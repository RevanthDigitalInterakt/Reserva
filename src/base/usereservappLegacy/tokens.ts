export const white = '#FFF';
export const fullBlack = '#000';

// cores primarias
export const preto = '#333333';
export const vermelhoFechadoRSV = '#5A0000';
export const marinho = '#13294B';
export const neutroFrio2 = '#8A8C8E';
export const neutroQuente2 = '#B1ACA8';
export const divider = '#D8D9DA';
export const dividerCounter = '#EAEAEA';
export const progressTextColor = '#979797';
export const searchBarTextColor = '#666666';

// cores secundarias
export const vermelhoRSV = '#A00000';
export const verde = '#22372B';
export const neutroQuente1 = '#D7CFCA';
export const offWhite = '#D6D1C4';
export const offWhite15 = '#F9F8F6';
export const whiteSecondary = '#FAFAFA';
export const neutroFrio1 = '#D8D9DA';
export const leve = '#F9E1DD';

// cores do sistema
export const vermelhoAlerta = '#EF1E1E';
export const verdeSucesso = '#38A238';
export const amareloAtencao = '#FEA528';
export const dropDownBorderColor = '#DDDDDD';

// larguras de borda
export const borderWidthNone = 0;
export const borderWidthHairline = 1;
export const borderWidthThin = 2;
export const borderWidthThick = 4;
export const borderWidthHeavy = 8;
export const borderDiscountTag = 12;

// cores de borda
export const borderButton = '#333333';
export const borderColorProduct = '#757575';

// backgrounds
export const backgroundMenuOpened = '#F1F1F1';
export const backgroundApp = '#F9F9F9';
export const modalBackDropColor = '#AFAFAF';
export const backgoundInput = '#F0F0F0';
export const backgoundDivider = '#F5F5F5';
