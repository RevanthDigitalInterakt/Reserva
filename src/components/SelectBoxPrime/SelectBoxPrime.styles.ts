import { StyleSheet, Dimensions } from 'react-native';
import { COLORS } from '../../base/styles/colors';

const smallDevices = Dimensions.get('screen').width <= 375;

export const styles = StyleSheet.create({
  checkBoxContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 16,
    borderWidth: 1,
    borderColor: COLORS.GOLD,
    backgroundColor: COLORS.BACKGROUND_GOLD,
  },
  minHeight: {
    minHeight: 72,
  },
  between: {
    justifyContent: 'space-between',
  },
  start: {
    justifyContent: 'flex-start',
  },
  ml: {
    marginLeft: 16,
  },
  imageBackgroundBadge: {
    width: smallDevices ? 135 : 160,
    height: smallDevices ? 24 : 28,
    justifyContent: 'center',
    paddingLeft: 10,
  },
  integerPartPrime: {
    fontWeight: '700',
    fontSize: 24,
    color: COLORS.DARK_GOLD_TEXT,
  },
  decimalPartPrime: {
    fontWeight: '700',
    color: COLORS.GOLD,
  },
  decimalText: {
    color:COLORS.DARK_GOLD_TEXT
  },
  normalTextRed: {
    fontWeight: '400',
    fontSize: 18,
    color: COLORS.DARK_GOLD_TEXT,
  },
  negativeMarginText: {
    marginLeft: smallDevices ? -35 : -40,
  },
  priceContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'baseline',
    marginTop: 4,
  },
  primePrice: {
    marginHorizontal: 16,
  },
  primeCheckBox: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: 3,
    height: 22,
    width: 22,
    borderRadius: 11,
    borderWidth: 1,
    borderColor: COLORS.GOLD,
    marginRight: smallDevices ? 15 : 0,
  },
  primeCheckBoxFill: {
    height: 16,
    width: 16,
    borderRadius: 8,
    backgroundColor: COLORS.BACKGROUND_GOLD,
  },
  priceDataWrapper: {
    alignItems: 'center',
    justifyContent: 'flex-end',
    paddingVertical: 10,
    marginLeft: 16,
  },
  priceDataEconomy: {
    alignItems: 'center',
    justifyContent: 'flex-end',
    paddingVertical: 10,
    marginLeft: 6,
  },
  separator: {
    alignSelf: 'center',
    width: 1,
    height: 30,
    backgroundColor: COLORS.LIGHT_GRAY,
    marginHorizontal: 15,
  },
  mt: {
    marginTop: 30,
  },
  mDefault: {
    margin: 0,
  },
  bePrimeBadge: {
    backgroundColor: COLORS.GOLD,
    width: 150,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 2,
  },
  textWhite: {
    color: COLORS.WHITE,
  },
  textRedBadge: {
    color: COLORS.WHITE,
    fontSize: smallDevices ? 12 : 15,
  },
});
