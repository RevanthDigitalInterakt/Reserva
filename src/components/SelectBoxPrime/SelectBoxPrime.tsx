import React, { useMemo } from 'react';
import {
  ImageBackground, StyleSheet, TouchableOpacity, View,
} from 'react-native';

import { useRemoteConfig, type TTypesInstallments } from '../../hooks/useRemoteConfig';
import { decimalPart, integerPart } from '../../utils/numberUtils';
import testProps from '../../utils/testProps';
import { Typography } from '../Typography/Typography';
import { styles } from './SelectBoxPrime.styles';
import EventProvider from '../../utils/EventProvider';

const goldBadgeImage = require('./assets/goldBadge.png');

interface SelectBoxPrimeProps {
  installment?: { number: number; value: number };
  isChecked: boolean;
  onPress: (option: string) => void;
  price?: number;
  savedValue: number;
}

export function SelectBoxPrime({
  installment,
  isChecked,
  onPress,
  price = 0,
  savedValue,
}: SelectBoxPrimeProps) {
  const { getString } = useRemoteConfig();

  const typeInstallments: TTypesInstallments = useMemo(() => (
    getString('installments_prime')
  ), [getString]) as TTypesInstallments;

  const containerSelectBoxes = StyleSheet.flatten([
    styles.checkBoxContainer,
    typeInstallments === 'hide_installments' ? styles.start : styles.between,
    typeInstallments === 'hide_installments' && styles.minHeight,
  ]);

  const containerPrices = StyleSheet.flatten([
    styles.priceDataWrapper,
    typeInstallments === 'hide_installments' && styles.ml,
  ]);

  const separator = StyleSheet.flatten([
    styles.separator,
    typeInstallments === 'hide_installments' ? styles.mDefault : styles.mt,
  ]);

  return (
    <TouchableOpacity
      style={styles.primePrice}
      onPress={() => {
        EventProvider.logEvent('prime_price_box_click', {});
        onPress('pricePrime');
      }}
      disabled={isChecked}
      {...testProps('select_box_price_prime')}
    >
      <View style={containerSelectBoxes}>
        <View style={styles.primeCheckBox}>
          {isChecked && (
            <View
              style={styles.primeCheckBoxFill}
              {...testProps('select_box_price_prime_checked')}
            />
          )}
        </View>

        <View style={containerPrices}>
          <Typography
            fontFamily="reservaSansRegular"
            style={styles.normalTextRed}
          >
            Clientes
            <Typography variant="descontoTag2" fontSize={20}>
              {' '}
              Prime
            </Typography>
          </Typography>

          {typeInstallments !== 'hide_installments' && installment && (
            <View style={styles.priceContainer}>
              <Typography
                fontFamily="reservaSansRegular"
                style={styles.normalTextRed}
              >
                {installment?.number}
                x
                {' '}
              </Typography>

              <Typography
                style={styles.integerPartPrime}
                fontFamily="reservaSansRegular"
              >
                R$
                {' '}
                {`${integerPart(installment.value)},`}
              </Typography>
              <Typography
                style={styles.decimalText}
                fontWeight="bold"
                fontFamily="reservaSansRegular"
              >
                {decimalPart(installment.value)}
              </Typography>
            </View>
          )}
        </View>

        <View style={separator} />

        {typeInstallments !== 'hide_installments' && (
          <View style={styles.priceDataEconomy}>
            <ImageBackground
              source={goldBadgeImage}
              resizeMode="cover"
              style={styles.imageBackgroundBadge}
            >
              <Typography
                fontFamily="reservaSansItalic"
                style={styles.textRedBadge}
              >
                Economize
                <Typography
                  fontFamily="reservaSansRegular"
                  style={{ fontWeight: 'bold' }}
                >
                  {' '}
                  R$
                  {savedValue.toFixed(2)}
                </Typography>
              </Typography>
            </ImageBackground>

            <View style={[styles.priceContainer, styles.negativeMarginText]}>
              <Typography
                style={styles.integerPartPrime}
                fontFamily="reservaSansRegular"
              >
                R$
                {' '}
                {`${integerPart(price)},`}
              </Typography>
              <Typography
                style={styles.decimalText}
                fontWeight="bold"
                fontFamily="reservaSansRegular"
              >
                {decimalPart(price)}
              </Typography>
            </View>
          </View>
        )}
        {typeInstallments === 'hide_installments' && (
          <>
            <Typography
              style={styles.integerPartPrime}
              fontFamily="reservaSansRegular"
            >
              R$
              {' '}
              {`${integerPart(price)},`}
            </Typography>
            <Typography
              style={styles.decimalText}
              fontWeight="bold"
              fontFamily="reservaSansRegular"
            >
              {decimalPart(price)}
            </Typography>
          </>
        )}
      </View>

      <View style={styles.bePrimeBadge}>
        <Typography fontFamily="reservaSansRegular" style={styles.textWhite}>
          Seja
          <Typography variant="descontoTag2" fontSize={18}>
            {' '}
            Prime
          </Typography>
        </Typography>
      </View>
    </TouchableOpacity>
  );
}
