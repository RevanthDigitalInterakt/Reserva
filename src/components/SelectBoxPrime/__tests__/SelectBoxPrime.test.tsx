import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { fireEvent, render, screen } from '@testing-library/react-native';
import { SelectBoxPrime } from '../SelectBoxPrime';
import { theme } from '../../../base/usereservappLegacy/theme';

const onPressFn = jest.fn();

const selectBoxPrimeMock = {
  installmentsNumber: 4,
  installmentsPrice: 64.5,
  isChecked: false,
  onPress: onPressFn,
  savedValue: 29,
  price: 149.9,
};

const TestingComponent = (
  <ThemeProvider theme={theme}>
    <SelectBoxPrime
      installmentsNumber={selectBoxPrimeMock.installmentsNumber}
      installmentsPrice={selectBoxPrimeMock.installmentsPrice}
      isChecked={selectBoxPrimeMock.isChecked}
      onPress={selectBoxPrimeMock.onPress}
      price={selectBoxPrimeMock.price}
      savedValue={selectBoxPrimeMock.savedValue}
    />
  </ThemeProvider>
);

describe('SelectBoxPrime', () => {
  beforeEach(() => render(TestingComponent));

  it('should render properly', () => {
    const selectBoxPrime = screen.getByTestId('com.usereserva:id/select_box_price_prime');

    expect(selectBoxPrime).toBeOnTheScreen();
  });

  it('should match with the snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should call onPressFn when select box is pressed', () => {
    const selectBox = screen.getByTestId('com.usereserva:id/select_box_price_prime');
    fireEvent.press(selectBox);

    expect(onPressFn).toHaveBeenCalled();
    expect(onPressFn).toHaveBeenCalledWith('pricePrime');
  });
});
