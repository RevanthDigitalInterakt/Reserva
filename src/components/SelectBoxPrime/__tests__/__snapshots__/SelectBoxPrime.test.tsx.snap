// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBoxPrime should match with the snapshot 1`] = `
<View
  accessibilityLabel="com.usereserva:id/select_box_price_prime"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginHorizontal": 16,
      "opacity": 1,
    }
  }
  testID="com.usereserva:id/select_box_price_prime"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF8FA",
        "borderColor": "#A01E21",
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#A01E21",
          "borderRadius": 11,
          "borderWidth": 1,
          "height": 22,
          "justifyContent": "center",
          "marginRight": 0,
          "padding": 3,
          "width": 22,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "flex-end",
          "marginLeft": 16,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        fontFamily="reservaSansRegular"
        style={
          Array [
            Object {
              "fontFamily": "ReservaSans-Regular",
            },
            Object {
              "color": "#A01E21",
              "fontSize": 18,
              "fontWeight": "400",
            },
          ]
        }
      >
        Clientes
        <Text
          fontSize={20}
          style={
            Array [
              Object {
                "fontFamily": "ReservaSerif-Bold",
                "fontSize": 20,
              },
            ]
          }
          variant="descontoTag2"
        >
           
          Prime
        </Text>
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#8D8D8D",
          "height": 30,
          "marginHorizontal": 15,
          "marginTop": 30,
          "width": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "flex-end",
          "marginLeft": 6,
          "paddingVertical": 10,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "height": 28,
            "justifyContent": "center",
            "paddingLeft": 10,
            "width": 160,
          }
        }
      >
        <Image
          resizeMode="cover"
          source={
            Object {
              "commons": Object {
                "GuideBermuda": [Circular],
                "GuideCalca": [Circular],
                "GuideCamiseta": [Circular],
                "GuideHoodie": [Circular],
                "GuideJaqueta": [Circular],
                "GuideMangaCurta": [Circular],
                "GuideMangaLonga": [Circular],
                "GuidePolo": [Circular],
                "GuideShort": [Circular],
                "GuideSueter": [Circular],
                "GuideSungaCueca": [Circular],
                "QrcodeBackground": [Circular],
                "americanExpress": [Circular],
                "arrowInstruction": [Circular],
                "bagEmpty": [Circular],
                "bannerCatalog": [Circular],
                "bannerLogin": [Circular],
                "bannerOffer": [Circular],
                "cardFront": [Circular],
                "cardVerso": [Circular],
                "clothingCare1": [Circular],
                "clothingCareBleach": [Circular],
                "clothingCareDrying": [Circular],
                "clothingCarePass": [Circular],
                "clothingCareProfessionalWashing": [Circular],
                "clothingCareWash": [Circular],
                "cupomModalBackground": [Circular],
                "dividerReserva": [Circular],
                "elo": [Circular],
                "emptyProductCatalog": [Circular],
                "facebook": [Circular],
                "foraDoAr": [Circular],
                "headerLogin": [Circular],
                "homeMock1": [Circular],
                "homeMock2": [Circular],
                "homeMock3": [Circular],
                "homeMock4": [Circular],
                "imageCashback": [Circular],
                "imageNotFound": [Circular],
                "instagram": [Circular],
                "localReserva": [Circular],
                "logoQRCode": [Circular],
                "mastercard": [Circular],
                "noStoresFound": [Circular],
                "noWishList": [Circular],
                "pinYou": [Circular],
                "purchaseConfirmation": [Circular],
                "raceImageBackground": [Circular],
                "saleOff": [Circular],
                "search": [Circular],
                "searchNotFound": [Circular],
                "selectRectangle": [Circular],
                "shirt1": [Circular],
                "shirt2": [Circular],
                "shirt3": [Circular],
                "shirt4": [Circular],
                "topBannerMock1": [Circular],
                "twitter": [Circular],
                "update": [Circular],
                "visa": [Circular],
                "whatsapp": [Circular],
                "withoutInternet": [Circular],
                "zoomHand": [Circular],
              },
              "default": Object {
                "GuideBermuda": [Circular],
                "GuideCalca": [Circular],
                "GuideCamiseta": [Circular],
                "GuideHoodie": [Circular],
                "GuideJaqueta": [Circular],
                "GuideMangaCurta": [Circular],
                "GuideMangaLonga": [Circular],
                "GuidePolo": [Circular],
                "GuideShort": [Circular],
                "GuideSueter": [Circular],
                "GuideSungaCueca": [Circular],
                "QrcodeBackground": [Circular],
                "americanExpress": [Circular],
                "arrowInstruction": [Circular],
                "bagEmpty": [Circular],
                "bannerCatalog": [Circular],
                "bannerLogin": [Circular],
                "bannerOffer": [Circular],
                "cardFront": [Circular],
                "cardVerso": [Circular],
                "chevronRight": [Function],
                "clothingCare1": [Circular],
                "clothingCareBleach": [Circular],
                "clothingCareDrying": [Circular],
                "clothingCarePass": [Circular],
                "clothingCareProfessionalWashing": [Circular],
                "clothingCareWash": [Circular],
                "cupomModalBackground": [Circular],
                "dividerReserva": [Circular],
                "elo": [Circular],
                "emptyProductCatalog": [Circular],
                "facebook": [Circular],
                "foraDoAr": [Circular],
                "headerLogin": [Circular],
                "homeMock1": [Circular],
                "homeMock2": [Circular],
                "homeMock3": [Circular],
                "homeMock4": [Circular],
                "imageCashback": [Circular],
                "imageNotFound": [Circular],
                "instagram": [Circular],
                "localReserva": [Circular],
                "logoPrime": [Function],
                "logoQRCode": [Circular],
                "mastercard": [Circular],
                "noStoresFound": [Circular],
                "noWishList": [Circular],
                "pinYou": [Circular],
                "purchaseConfirmation": [Circular],
                "raceImageBackground": [Circular],
                "saleOff": [Circular],
                "search": [Circular],
                "searchNotFound": [Circular],
                "selectRectangle": [Circular],
                "shirt1": [Circular],
                "shirt2": [Circular],
                "shirt3": [Circular],
                "shirt4": [Circular],
                "topBannerMock1": [Circular],
                "twitter": [Circular],
                "update": [Circular],
                "visa": [Circular],
                "whatsapp": [Circular],
                "withoutInternet": [Circular],
                "zoomHand": [Circular],
              },
              "svgs": Object {
                "chevronRight": [Function],
                "logoPrime": [Function],
              },
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 28,
                "width": 160,
              },
              undefined,
            ]
          }
        />
        <Text
          fontFamily="reservaSansItalic"
          style={
            Array [
              Object {
                "fontFamily": "ReservaSans-Italic",
              },
              Object {
                "color": "#ffffff",
                "fontSize": 15,
              },
            ]
          }
        >
          Economize
          <Text
            fontFamily="reservaSansRegular"
            style={
              Array [
                Object {
                  "fontFamily": "ReservaSans-Regular",
                },
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          >
             
            R$
            29.00
          </Text>
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "baseline",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginTop": 4,
            },
            Object {
              "marginLeft": -40,
            },
          ]
        }
      >
        <Text
          fontFamily="reservaSansRegular"
          style={
            Array [
              Object {
                "fontFamily": "ReservaSans-Regular",
              },
              Object {
                "color": "#A01E21",
                "fontSize": 24,
                "fontWeight": "700",
              },
            ]
          }
        >
          R$
           
          149,
        </Text>
        <Text
          color="vermelhoRSV"
          fontFamily="reservaSansRegular"
          fontWeight="bold"
          style={
            Array [
              Object {
                "color": "#A00000",
                "fontFamily": "ReservaSans-Regular",
                "fontWeight": "700",
              },
            ]
          }
        >
          90
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#A01E21",
        "justifyContent": "center",
        "padding": 2,
        "width": 150,
      }
    }
  >
    <Text
      fontFamily="reservaSansRegular"
      style={
        Array [
          Object {
            "fontFamily": "ReservaSans-Regular",
          },
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      Seja
      <Text
        fontSize={18}
        style={
          Array [
            Object {
              "fontFamily": "ReservaSerif-Bold",
              "fontSize": 18,
            },
          ]
        }
        variant="descontoTag2"
      >
         
        Prime
      </Text>
    </Text>
  </View>
</View>
`;
